### PR TITLE
CI setup

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @willow-ahrens @hameerabbasi,

This PR adds a simple CI setup for running tests. First #2 needs to be merged.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+          key:
+            test-${{ matrix.os }}-conda-py${{ matrix.python }}-${{ env.CACHE_NUMBER }}-${{
+            hashFiles('ci/environment.yml') }}
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          activate-environment: finch-tensor-dev
+          allow-softlinks: true
+          environment-file: ci/environment.yml
+          python-version: ${{ matrix.python }}
+          miniforge-variant: Mambaforge
```
Comment (hameerabbasi) [inline .github/workflows/ci.yml]: I'd change this to `Miniforge` as Mambaforge is deprecated.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+jobs:
+  test:
+    defaults:
+      run:
+        shell: bash -el {0}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python: ['3.8', '3.9', '3.10', '3.11']
```
Comment (hameerabbasi) [inline .github/workflows/ci.yml]:
```suggestion
        python: ['3.9', '3.10', '3.11']
```

I'd drop Py3.8, it's not supported by NumPy (see [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html)).

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        include:
+          - os: macos-latest
+            python: '3.10'
```
Comment (hameerabbasi) [inline .github/workflows/ci.yml]:
```suggestion
            python: '3.9'
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+          - os: windows-latest
+            python: '3.10'
```
Comment (hameerabbasi) [inline .github/workflows/ci.yml]:
```suggestion
            python: '3.9'
```

Testing against the oldest version usually reveals more bugs.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        if: always()
+        with:
+          files: "**/test-*.xml"
+on:
+  # Trigger the workflow on push or pull request,
+  # but only for the main branch
+  push:
+    branches:
+      - main
+      - vnext
```
Comment (hameerabbasi) [inline .github/workflows/ci.yml]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  pull_request:
+    branches:
+      - main
+      - vnext
```
Comment (hameerabbasi) [inline .github/workflows/ci.yml]:
```suggestion
```

**Review by hameerabbasi — COMMENTED** [review]
Except the `vnext` change, everything should be ported to `sparse` at some point.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mtsokol) [reply .github/workflows/ci.yml]: Done!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (mtsokol) [reply .github/workflows/ci.yml]: Done!

File: .github/workflows/ci.yml (hunk 3 above)
Comment (mtsokol) [reply .github/workflows/ci.yml]: Done!

File: .github/workflows/ci.yml (hunk 4 above)
Comment (mtsokol) [reply .github/workflows/ci.yml]: Done!

File: .github/workflows/ci.yml (hunk 5 above)
Comment (mtsokol) [reply .github/workflows/ci.yml]: Done!

File: .github/workflows/ci.yml (hunk 6 above)
Comment (mtsokol) [reply .github/workflows/ci.yml]: Done!

**Review by hameerabbasi — APPROVED** [review]

**Comment by willow-ahrens** [comment]
This doesn't seem to work. Is the reason the tests are failing the stated dependency on #2?

**Comment by willow-ahrens** [comment]
I don't know how to use setuptools, or setuptools-scm, so I'm a little confused about how to proceed and it'll take me a few days to get used to it.

**Pull request merged**